ckan/public/base/javascript/modules/image-upload.js

```diff
@@ -105,25 +110,71 @@ this.ckan.module('image-upload', function($, _) {
 
       if (options.is_url) {
         this._showOnlyFieldUrl();
+
+        this._updateUrlLabel(this.i18n('label_for_url'));
       } else if (options.is_upload) {
         this._showOnlyFieldUrl();
+
         this.field_url_input.prop('readonly', true);
+        // If the data is an uploaded file, the filename will display rather than whole url of the site
+        var filename = this._fileNameFromUpload(this.field_url_input.val());
+        this.field_url_input.val(filename);
+
+        this._updateUrlLabel(this.i18n('label_for_upload'));
       } else {
         this._showOnlyButtons();
       }
     },
 
+    /* Quick way of getting just the filename from the uri of the resource data
+     *
+     * url - The url of the uploaded data file
+     *
+     * Returns String.
+     */
+    _fileNameFromUpload: function(url) {
```

Comment:
We could replace this function with this regex `/[^\/?#]+(?=$|[?#])/` taken from [stackoverflow](http://stackoverflow.com/a/6543380/4099396). It would make the code shorter but as it is regex also harder to read. Your call @jrods.